### PR TITLE
Fix 12188

### PR DESCRIPTION
**Pull request by githubjeka**
| Fixed issues  | #12188
- [X] \yii\caching\Cache
- [x] Review implementations of \yii\caching\Cache
- [x] Review documentation (nothing had changed)

### CODE REVIEW

File: framework/caching/DbCache.php [hunk 1]
```diff
@@ -179,7 +179,7 @@ protected function getValues($keys)
      * This is the implementation of the method declared in the parent class.
      *
      * @param string $key the key identifying the value to be cached
-     * @param string $value the value to be cached
+     * @param string $value the value to be cached. Other types(if you disabled [[serializer]]) cannot be saved.
```
Comment (SilverFire) [inline framework/caching/DbCache.php]: Missing space before `(`

File: framework/caching/DbCache.php [hunk 2]
```diff
@@ -205,7 +205,7 @@ protected function setValue($key, $value, $duration)
...
```
Comment (SilverFire) [inline framework/caching/DbCache.php]: same here

File: framework/caching/DbCache.php (hunk 1 above)
Comment (githubjeka) [inline framework/caching/DbCache.php]: Ok, pls review all. And I send fixes

File: framework/caching/FileCache.php [hunk 3]
```diff
@@ -127,7 +127,8 @@ protected function getValue($key)
      * This is the implementation of the method declared in the parent class.
      *
      * @param string $key the key identifying the value to be cached
-     * @param string $value the value to be cached
+     * @param string $value the value to be cached. Other types(if you disabled [[serializer]]) unable to get is
```
Comment (SilverFire) [inline framework/caching/FileCache.php]: Missing space before `(`

Comment (SilverFire) [inline framework/caching/FileCache.php]: The sentence looks inconsistentюю

File: framework/caching/FileCache.php [hunk 4]
```diff
@@ -159,7 +160,8 @@ protected function setValue($key, $value, $duration)
...
```
Comment (SilverFire) [inline framework/caching/FileCache.php]: Same here

File: framework/caching/FileCache.php (hunk 3 above)
Comment (SilverFire) [inline framework/caching/FileCache.php]: As for me, `@param string $value the value to be cached` is enough.

File: framework/caching/DbCache.php (hunk 1 above)
Comment (SilverFire) [inline framework/caching/DbCache.php]: As for me, `@param string $value the value to be cached` is enough.

File: framework/caching/Cache.php [hunk 5]
```diff
@@ -386,7 +388,8 @@ public function flush()
      * This method should be implemented by child classes to store the data
      * in specific cache storage.
      * @param string $key the key identifying the value to be cached
-     * @param string $value the value to be cached
+     * @param mixed $value the value to be cached. Most often it's a string. If you disabled [[serializer]],
```
Comment (SilverFire) [inline framework/caching/Cache.php]: `If you have disabled`

File: framework/caching/Cache.php [hunk 6]
```diff
@@ -377,7 +378,8 @@ public function flush()
      * This method should be implemented by child classes to retrieve the data
      * from specific cache storage.
      * @param string $key a unique key identifying the cached value
-     * @return string|boolean the value stored in cache, false if the value is not in the cache or expired.
+     * @return mixed|false the value stored in cache, false if the value is not in the cache or expired. Most often
+     * value is a string. If you disabled [[serializer]], it could be something else.
```
Comment (SilverFire) [inline framework/caching/Cache.php]: `If you have disabled`

File: framework/caching/ApcCache.php [hunk 7]
```diff
@@ -115,7 +116,8 @@ protected function setValues($data, $duration)
      * Stores a value identified by a key into cache if the cache does not contain this key.
      * This is the implementation of the method declared in the parent class.
      * @param string $key the key identifying the value to be cached
-     * @param string $value the value to be cached
+     * @param mixed $value the value to be cached. Most often it's a string. If you disabled [[serializer]],
```
Comment (SilverFire) [inline framework/caching/ApcCache.php]: `If you have disabled`

File: framework/caching/ApcCache.php [hunk 8]
```diff
@@ -90,9 +90,10 @@ protected function getValues($keys)
      * This is the implementation of the method declared in the parent class.
      *
      * @param string $key the key identifying the value to be cached
-     * @param string $value the value to be cached
+     * @param mixed $value the value to be cached. Most often it's a string. If you disabled [[serializer]],
```
Comment (SilverFire) [inline framework/caching/ApcCache.php]: `If you have disabled`

**Comment by SilverFire** [comment]
Спасибо

**Pull request merged**